### PR TITLE
Fixes for viewing all dims of array or stack

### PR DESCRIPTION
**Pull request by sethaxen**
This PR fixes the issues with `view` raised in #423

There are some errors in the `single-arity views` testset that I don't understand yet.

### CODE REVIEW

**Comment by sethaxen** [comment]
The failing tests look broken. `idx` is set for each loop iteration but then always set to an `Int` internally:
https://github.com/rafaqz/DimensionalData.jl/blob/70f2aefc6c4248a0ab957f51cf8b415f9799a39e/test/selector.jl#L821-L837

**Comment by sethaxen** [comment]
The only remaining error seems to be related to uploading the coverage report to Codecov on OS X, e.g. https://github.com/rafaqz/DimensionalData.jl/actions/runs/3560347385/jobs/5980312809#step:7:47

**Comment by rafaqz** [comment]
Yeah that's happening everywhere now, but not sure what the issue is.

**Comment by rafaqz** [comment]
You may need to rebase this against main for the codecov fix

File: test/selector.jl [hunk 1]
```diff
@@ -1016,24 +1016,6 @@ end
         end
     end
 
-    @testset "single-arity views" begin
```
Comment (rafaqz) [inline test/selector.jl:1019]: Are we losing code coverage from deleting these?

**Review by rafaqz — COMMENTED** [review]
LGTM, just wondering if deleting those tests is a good ides...

File: test/selector.jl (hunk 1 above)
Comment (sethaxen) [reply test/selector.jl]: It doesn't seem so (diff coverage is 100% and total coverage is increased), but I've reverted the removal and instead fixed the tests.

**Review by rafaqz — APPROVED** [review]
Thanks for the PR!

**Pull request merged**